src/CentralDesktop/Parallel/ForkManager.php

```diff
@@ -137,9 +137,14 @@ function start($id = "Some Random Process") {
                      */
                     $status   = 0;
                     $childpid = pcntl_wait($status, WNOHANG);
+                    // If nothing happened, or an error happened, send out signals and delay
                     if ($childpid === 0) {
                         pcntl_signal_dispatch();
                         time_nanosleep(0, 500000000); // Sleep for 0.5s
+                    } elseif ($childpid === -1) {
+                        // Go ahead and dispatch messages and sleep.  We don't want to busy wait.
+                        pcntl_signal_dispatch();
+                        time_nanosleep(0, 500000000); // Sleep for 0.5s
```

Comment:
This is just a nice to have.  In the event that waitpid() returns -1, this clause prevents the code from going to 100% CPU usage.